packages/@glimmer/runtime/test/modifiers-test.ts

```diff
@@ -78,16 +80,223 @@ class ModifierTests extends RenderTest {
+    );
+    this.rerender({ something: 'another', foo: 'thingy' });
+    assert.deepEqual(modifierParams, ['another']);
+    assert.deepEqual(modifierNamedArgs, { foo: 'thingy' });
+    assert.equal(
+      modifiedElement && modifiedElement.getAttribute('id'),
+      'inner-div',
+      'Modifier is called on the element receiving the splattributes'
+    );
+  }
+
+  @test
+  'modifiers on components accept `this` in both positional params and named arguments, and updates when it changes'(
+    assert: Assert
+  ) {
+    let modifierParams: Option<unknown[]> = null;
+    let modifierNamedArgs: Option<Dict<unknown>> = null;
+    let modifiedElement: SimpleElement | undefined;
+    class Bar extends AbstractInsertable {
+      didInsertElement(params: unknown[], namedArgs: Dict<unknown>) {
+        modifierParams = params;
+        modifierNamedArgs = namedArgs;
+        modifiedElement = this.element;
+      }
+      didUpdate(params: unknown[], namedArgs: Dict<unknown>) {
+        modifierParams = params;
+        modifierNamedArgs = namedArgs;
+        modifiedElement = this.element;
+      }
+    }
+    let context = { id: 1 };
+    let context2 = { id: 2 };
+    this.registerComponent('Glimmer', 'TheFoo', '<div id="inner-div" ...attributes>Foo</div>');
+    this.registerModifier('bar', Bar);
+    this.render('<TheFoo {{bar "name" this foo=this}}/>', context);
+    assert.deepEqual(modifierParams, ['name', context]);
+    assert.deepEqual(modifierNamedArgs, { foo: context });
+    assert.equal(
+      modifiedElement && modifiedElement.getAttribute('id'),
+      'inner-div',
+      'Modifier is called on the element receiving the splattributes'
+    );
+    this.rerender(context2);
+    assert.deepEqual(modifierParams, ['name', context2]);
+    assert.deepEqual(modifierNamedArgs, { foo: context2 });
+    assert.equal(
+      modifiedElement && modifiedElement.getAttribute('id'),
+      'inner-div',
+      'Modifier is called on the element receiving the splattributes'
+    );
+  }
+
+  @test
+  'modifiers on components accept local variables in both positional params and named arguments, and updates when they change'(
+    assert: Assert
+  ) {
+    let modifierParams: Option<unknown[]> = null;
+    let modifierNamedArgs: Option<Dict<unknown>> = null;
+    let modifiedElement: SimpleElement | undefined;
+    class Bar extends AbstractInsertable {
+      didInsertElement(params: unknown[], namedArgs: Dict<unknown>) {
+        modifierParams = params;
+        modifierNamedArgs = namedArgs;
+        modifiedElement = this.element;
+      }
+      didUpdate(params: unknown[], namedArgs: Dict<unknown>) {
+        modifierParams = params;
+        modifierNamedArgs = namedArgs;
+        modifiedElement = this.element;
+      }
+    }
+    this.registerComponent('Glimmer', 'TheFoo', '<div id="inner-div" ...attributes>Foo</div>');
+    this.registerModifier('bar', Bar);
+    this.render(
+      `
+      {{#with this.foo as |v|}}
```

Comment:
I had to change `{{#let}}` to `{{#with}}` on this test because `let` doesn't seem to exist in this version of glimmer. What do you suggest we do here? Leave `with` in this branch alone or come up with a single testing approach that works in both branches?

Reply:
with is totally fine 👍